addons/project/tests/test_project_subtasks.py

```diff
@@ -0,0 +1,160 @@
+        self.assertFalse(self.task_1.child_ids.display_project_id, "The display project of a sub task should be false to project_id.")
+
+        with Form(self.task_1.with_context({'tracking_disable': True})) as task_form:
+            with task_form.child_ids.edit(0) as subtask_form:
+                subtask_form.display_project_id = self.project_goats
+
+        self.assertEqual(self.task_1.project_id, self.project_pigs, "Changing the project of a subtask should not change parent project")
+        self.assertEqual(self.task_1.child_ids.display_project_id, self.project_goats, "Display Project of the task should be well assigned")
+        self.assertEqual(self.task_1.child_ids.project_id, self.project_goats, "Changing display project id on a subtask should change project id")
+
+        with Form(self.task_1.with_context({'tracking_disable': True})) as task_form:
+            with task_form.child_ids.edit(0) as subtask_form:
+                subtask_form.display_project_id = self.env['project.project']
+
+        self.assertFalse(self.task_1.child_ids.display_project_id, "Display Project of the task should be well assigned, to False")
+        self.assertEqual(self.task_1.child_ids.project_id, self.project_pigs, "Resetting display project to False on a subtask should change project id to parent project id")
+
+        with Form(self.task_1.with_context({'tracking_disable': True})) as task_form:
+            task_form.project_id = self.project_goats
+
+        self.assertEqual(self.task_1.project_id, self.project_goats, "Parent project should change.")
+        self.assertFalse(self.task_1.child_ids.display_project_id, "Display Project of the task should be False")
+        self.assertEqual(self.task_1.child_ids.project_id, self.project_goats, "Resetting display project to False on a subtask should follow project of its parent")
+
+        with Form(self.task_1.with_context({'tracking_disable': True})) as task_form:
+            with task_form.child_ids.edit(0) as subtask_form:
+                subtask_form.display_project_id = self.project_goats
+            task_form.project_id = self.project_pigs
+
+        self.assertEqual(self.task_1.project_id, self.project_pigs, "Parent project should change back.")
+        self.assertEqual(self.task_1.child_ids.display_project_id, self.project_goats, "Display Project of the task should be well assigned")
+        self.assertEqual(self.task_1.child_ids.project_id, self.project_goats, "Changing display project id on a subtask should change project id")
+
+        with self.assertRaises(AssertionError):
+            with Form(self.task_1.child_ids.with_context({'tracking_disable': True})) as subtask_form:
+                subtask_form.display_project_id = self.env['project.project']
+                subtask_form.parent_id = self.env['project.task']
+                # Should raise an AssertionError as we don't reset the project
```

Comment:
The error is raised because the subtask has no longer any parent, isn't it ? 🤔

Reply:
You're right ! :) As we do not longer have a parent, we no longer have the computed project_id, so it becomes False, and the Form requires the project_id to be defined (not False).

Do you think I have to be more specific ?

Reply:
Okay, great, thanks. I think no, it is clear. It was just to see if I correctly understood this test 🙂